docs/sequence.puml

```diff
@@ -0,0 +1,58 @@
+@startuml
+database Bitcoin
+actor Bob
+Actor Alice
+database Monero
+
+Bob --> Alice: Establish connection
+
+group Request Response Channel
+    Bob -> Alice: Request Amounts
+    note left: Btc Amount
+
+    Alice --> Bob: Send Amounts
+    note right: Btc & Xmr Amounts
+end
+
+group Request Response Channel
+    Bob -> Alice: bob::Message0
+    note left: Pubkeys\ndleq proof sb\nxmr viewkey vb\nbtc refund addr
+
+    Alice --> Bob: alice::Message0
+    note right: Pubkeys\ndleq proof sa\nxmr view key va\nbtc redeem addr\nbtc punish addr
+end
+
+group Request Response Channel
+    Bob -> Alice: bob::Message1
+    note left: btc lock tx
+
+    Alice --> Bob: alice::Message1
+    note right: cancel tx sig\ntx refund enc sig
+end
+
+group Request Response Channel
+    Bob -> Alice: bob::Message2
+    note left: punish tx sig\ncancel tx sig
+
+== Negotiated ==
```

Comment:
What does this mean please? (:

Reply:
It means that negotiation phase ended. will replace.

Reply:
Interesting, for me the negotiation phase ends after the rate and amounts are exchanged (or before the first protocol-related message was exchanged).

Reply:
Indeed, it looks like we have introduced a terminology in the code that does not match what we previously agree in terms of negotiation vs execution. I will propose a new terminology as part this exercise.